parachains/runtimes/bridge-hubs/test-utils/src/test_cases.rs

```diff
@@ -392,123 +346,598 @@ pub fn message_dispatch_routing_works<
+	+ pallet_balances::Config
+	+ pallet_utility::Config
+	+ pallet_session::Config
+	+ pallet_xcm::Config
+	+ parachain_info::Config
+	+ pallet_collator_selection::Config
+	+ cumulus_pallet_dmp_queue::Config
+	+ cumulus_pallet_parachain_system::Config
+	+ cumulus_pallet_xcmp_queue::Config
+	+ pallet_bridge_grandpa::Config<GPI>
+	+ pallet_bridge_parachains::Config<PPI>
+	+ pallet_bridge_messages::Config<MPI, InboundPayload = XcmAsPlainPayload>
+	+ pallet_bridge_relayers::Config,
+	GPI: 'static,
+	PPI: 'static,
+	MPI: 'static,
+	MB: MessageBridge,
+	<MB as MessageBridge>::BridgedChain: Send + Sync + 'static,
+	UnderlyingChainOf<MessageBridgedChain<MB>>: bp_runtime::Chain<Hash = ParaHash> + Parachain,
+	XcmConfig: xcm_executor::Config,
+	HrmpChannelOpener: frame_support::inherent::ProvideInherent<
+		Call = cumulus_pallet_parachain_system::Call<Runtime>,
+	>,
+	ValidatorIdOf<Runtime>: From<AccountIdOf<Runtime>>,
+	<<Runtime as pallet_bridge_messages::Config<MPI>>::SourceHeaderChain as SourceHeaderChain>::MessagesProof: From<FromBridgedChainMessagesProof<ParaHash>>,
+	<<Runtime as pallet_bridge_grandpa::Config<GPI>>::BridgedChain as bp_runtime::Chain>::Hash: From<ParaHash>,
+	ParaHash: From<<<Runtime as pallet_bridge_grandpa::Config<GPI>>::BridgedChain as bp_runtime::Chain>::Hash>,
+	<Runtime as frame_system::Config>::AccountId:
+	Into<<<Runtime as frame_system::Config>::RuntimeOrigin as OriginTrait>::AccountId>,
+	AccountIdOf<Runtime>: From<sp_core::sr25519::Public>,
+	<Runtime as pallet_bridge_messages::Config<MPI>>::InboundRelayer: From<AccountId32>,
+	<Runtime as pallet_utility::Config>::RuntimeCall:
+	From<pallet_bridge_grandpa::Call<Runtime, GPI>>
+	+ From<pallet_bridge_parachains::Call<Runtime, PPI>>
+	+ From<pallet_bridge_messages::Call<Runtime, MPI>>
+{
+	assert_ne!(runtime_para_id, sibling_parachain_id);
+	assert_ne!(runtime_para_id, bridged_para_id);
+
+	// Relayer account at local/this BH.
+	let relayer_at_target = Bob;
+	let relayer_id_on_target: AccountIdOf<Runtime> = relayer_at_target.public().into();
+	let relayer_initial_balance = existential_deposit * 100000u32.into();
+	// Relayer account at remote/bridged BH.
+	let relayer_at_source = Dave;
+	let relayer_id_on_source: AccountId32 = relayer_at_source.public().into();
+
+	ExtBuilder::<Runtime>::default()
+		.with_collators(collator_session_key.collators())
+		.with_session_keys(collator_session_key.session_keys())
+		.with_safe_xcm_version(XCM_VERSION)
+		.with_para_id(runtime_para_id.into())
+		.with_balances(vec![(relayer_id_on_target.clone(), relayer_initial_balance)])
+		.with_tracing()
+		.build()
+		.execute_with(|| {
+			let zero: <Runtime as frame_system::Config>::BlockNumber = 0u32.into();
+			let genesis_hash = frame_system::Pallet::<Runtime>::block_hash(zero);
+			let mut header: <Runtime as frame_system::Config>::Header =
+				bp_test_utils::test_header(1u32.into());
+			header.set_parent_hash(genesis_hash);
+			executive_init_block(&header);
```

Comment:
@acatangiu just out of curiosity, do we need this?
```
			let zero: <Runtime as frame_system::Config>::BlockNumber = 0u32.into();
			let genesis_hash = frame_system::Pallet::<Runtime>::block_hash(zero);
			let mut header: <Runtime as frame_system::Config>::Header =
				bp_test_utils::test_header(1u32.into());
			header.set_parent_hash(genesis_hash);
			executive_init_block(&header);
```

I tried to remove this lines and `executive_init_block` and test is working,
so probably all the related stuff to executive_init_block could be removed